### PR TITLE
Log should_process reason and add test

### PR DESCRIPTION
**Pull request by fernandog**
Re-sending
@p0psicles i did some readings and its ok to return status like this.

Anyway this may be removed soon with new quality system.
At least now we have tests and better logging.

### CODE REVIEW

File: tests/test_should_process.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        'preferred_qualities': [Quality.HDWEBDL, Quality.HDBLURAY],
+        'expected': True
+    },
+    {  # p5: New quality not in quality system: no
+        'cur_quality': Quality.HDTV,
+        'new_quality': Quality.SDTV,
+        'allowed_qualities': [Quality.HDTV],
+        'preferred_qualities': [Quality.HDWEBDL, Quality.HDBLURAY],
+        'expected': False
+    },
```
Comment (fernandog) [inline tests/test_should_process.py]: @OmgImAlexis @duramato any other scenario that I'm missing?

File: tests/test_should_process.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {  # p6: Preferred lower quality replacing current higher allowed: yes
+        'cur_quality': Quality.HDWEBDL,
+        'new_quality': Quality.HDTV,
+        'allowed_qualities': [Quality.HDWEBDL],
+        'preferred_qualities': [Quality.HDTV],
+        'expected': True
+    },
+])
```
Comment (fernandog) [inline tests/test_should_process.py]: @OmgImAlexis @duramato any other scenario that I'm missing?

File: medusa/post_processor.py [hunk 3]
```diff
@@ -856,22 +855,37 @@ def _should_process(current_quality, new_quality, allowed, preferred):
...
+
         :param current_quality: The current quality of the episode that is being processed
         :param new_quality: The new quality of the episode that is being processed
         :param allowed: Qualities that are allowed
         :param preferred: Qualities that are preferred
-        :return: True if the quality should be processed, False or None otherwise.
+        :return: Tuple with Boolean if the quality should be processed and String with reason if should process or not
         """
+        if current_quality is common.Quality.NONE:
+            return False, 'There is no current quality. Skipping as we can only replace existing qualities'
```
Comment (medariox) [inline medusa/post_processor.py]: This is useless and should be removed.

File: medusa/post_processor.py [hunk 4]
```diff
@@ -856,22 +855,37 @@ def _should_process(current_quality, new_quality, allowed, preferred):
...
+                    return False, 'New quality is same|lower than current Preferred. Ignoring quality'
+            return True, 'New quality is Preferred'
         elif new_quality in allowed:
             if current_quality in preferred:
-                return False
+                return False, 'Current quality is Allowed but we already have a current Preferred. Ignoring quality'
             elif current_quality not in allowed:
-                return True
-            return new_quality > current_quality
+                return True, "New quality is Allowed and we don\'t have a current Preferred. Accepting quality"
```
Comment (medariox) [inline medusa/post_processor.py]: No ```\``` needed here.

File: medusa/post_processor.py [hunk 5]
```diff
@@ -1013,12 +1027,17 @@ def process(self):
                     self._log(u'New file is a PROPER, marking it safe to replace')
                     self.flag_kodi_clean_library()
                 else:
-                    allowed, preferred = show.current_qualities
-                    if not self._should_process(old_ep_quality, new_ep_quality, allowed, preferred):
+                    allowed_qualities, preferred_qualities = show.current_qualities
+                    self._log(u'Checking if new quality {0} should reaplace current quality: {1}'.format
```
Comment (medariox) [inline medusa/post_processor.py]: ```replace```

**Comment by p0psicles** [comment]
@fernandog i think all medariox comments are valid, can you create a new PR, to resolve these?

File: medusa/post_processor.py (hunk 3 above)
Comment (fernandog) [inline medusa/post_processor.py]: Its not useless. It make sure it will run only when there is valid quality. Fail safe.

Comment (medariox) [inline medusa/post_processor.py]: How can we have an "invalid" quality? If the quality is "invalid" it will be Unknown.

Comment (fernandog) [inline medusa/post_processor.py]: did you check the code before you say that?

Comment (medariox) [inline medusa/post_processor.py]: Yes. Like I said, it's useless and needs to be removed.

**Pull request merged**